testsys/src/run_vmware.rs

```diff
@@ -0,0 +1,255 @@
+use crate::error::{self, Result};
+use bottlerocket_agents::sonobuoy::Mode;
+use bottlerocket_agents::wireguard::WIREGUARD_SECRET_NAME;
+use bottlerocket_agents::{
+    K8sVersion, SonobuoyConfig, VSphereVmConfig, AWS_CREDENTIALS_SECRET_NAME,
+    VSPHERE_CREDENTIALS_SECRET_NAME,
+};
+use kube::{api::ObjectMeta, Client};
+use maplit::btreemap;
+use model::clients::{CrdClient, ResourceClient, TestClient};
+use model::constants::NAMESPACE;
+use model::{
+    Agent, Configuration, DestructionPolicy, Resource, ResourceSpec, SecretName, Test, TestSpec,
+};
+use snafu::ResultExt;
+use std::fs::read_to_string;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+/// Create an EKS resource, EC2 resource and run Sonobuoy.
+#[derive(Debug, StructOpt)]
+pub(crate) struct RunVmware {
+    /// Name of the sonobuoy test.
+    #[structopt(long, short)]
+    name: String,
+
+    /// Location of the sonobuoy test agent image.
+    // TODO - default to an ECR public repository image
+    #[structopt(long, short)]
+    test_agent_image: String,
+
+    /// Name of the pull secret for the sonobuoy test image (if needed).
+    #[structopt(long)]
+    test_agent_pull_secret: Option<String>,
+
+    /// Keep the test agent running after completion.
+    #[structopt(long)]
+    keep_running: bool,
+
+    /// The plugin used for the sonobuoy test. Normally this is `e2e` (the default).
+    #[structopt(long, default_value = "e2e")]
+    sonobuoy_plugin: String,
+
+    /// The mode used for the sonobuoy test. One of `non-disruptive-conformance`,
+    /// `certified-conformance`, `quick`. Although the Sonobuoy binary defaults to
+    /// `non-disruptive-conformance`, we default to `quick` to make a quick test the most ergonomic.
+    #[structopt(long, default_value = "quick")]
+    sonobuoy_mode: Mode,
+
+    /// The kubernetes version (with or without the v prefix). Examples: v1.21, 1.21.3, v1.20.1
+    #[structopt(long)]
+    kubernetes_version: Option<K8sVersion>,
+
+    /// The kubernetes conformance image used for the sonobuoy test.
+    #[structopt(long)]
+    kubernetes_conformance_image: Option<String>,
+
+    /// The name of the secret containing aws credentials.
+    #[structopt(long)]
+    aws_secret: SecretName,
+
+    /// The name of the secret containing vsphere credentials.
+    #[structopt(long)]
+    vsphere_secret: SecretName,
+
+    /// The name of the secret containing wireguard credentials.
+    #[structopt(long)]
+    wireguard_secret: SecretName,
+
+    /// The name of the vsphere cluster that will be used (whether it is being created or already
+    /// exists).
```

Comment:
```suggestion
    /// The name of the vsphere cluster that will be used.
```

I think it has to be created already since testsys doesn't provide the means to create vmware clusters as of yet,